whisper.py

```diff
@@ -48,72 +48,77 @@ def load_model(model_base_dir: str, model_type: str, device: str) -> WhisperMode
     return model
 
 
-def run_asr(input_path, output_dir, model=None) -> Optional[dict]:
+def run_asr(input_path, output_dir, model=None) -> dict | str:
     logger.info(f"Starting ASR on {input_path}")
     start_time = time.time()
-    if not model:
-        logger.info("Model not passed as param, need to obtain it first")
-        model = load_model(model_base_dir, w_model, w_device)
-    logger.info("Processing segments")
-    segments, _ = model.transcribe(
-        input_path,
-        vad_filter=w_vad,
-        beam_size=w_beam_size,
-        best_of=w_best_of,
-        # temperature=ast.literal_eval(w_temperature),
-        language="nl",
-        word_timestamps=w_word_timestamps,
-    )
-
-    segments_to_add = []
-    for segment in segments:
-        words_to_add = []
-        if w_word_timestamps:
-            for word in segment.words:
-                words_to_add.append(
-                    {
-                        "text": word.word.strip(),
-                        "start": word.start,
-                        "end": word.end,
-                        "confidence": word.probability,
-                    }
-                )
-        segments_to_add.append(
-            {
-                "id": segment.id,
-                "seek": segment.seek,
-                "start": segment.start,
-                "end": segment.end,
-                "text": segment.text.strip(),
-                "tokens": segment.tokens,
-                "temperature": segment.temperature,
-                "avg_logprob": segment.avg_logprob,
-                "compression_ratio": segment.compression_ratio,
-                "no_speech_prob": segment.no_speech_prob,
-                "words": words_to_add,
-            }
+    # I wanted to add more detailed errors, but function becomes too complex...
```

Comment:
I think if you use Python's exception mechanism and raise an exception with a message rather than return a string, it may become easier to provide error details while keeping the code robust

Reply:
yes definitely agree. Also makes unit tests very clear to read (in respect to error handling; also you can force raising the exceptions to test if the code is catching the exceptions properly)

Reply:
You are right, it'd be more robust to use the exception mechanism. I'll look more into how to properly implement it and work on that and hopefully finish it next week!